src/core/random.hpp

```diff
@@ -101,38 +101,31 @@ Utils::Vector<uint64_t, 4> philox_4_uint64s(uint64_t counter, int key1,
  * not correlated between two calls along the same counter
  * sequence.
  *
- */
-template <RNGSalt salt> double noise(uint64_t counter, int key1, int key2 = 0) {
-
-  auto const noise = philox_4_uint64s<salt>(counter, key1, key2);
-
-  using Utils::uniform;
-  return uniform(noise[0]) - 0.5;
-}
-
-/**
- * @brief 3d uniform vector noise.
- *
- * Mean = 0, variance = 1 / 12.
- * This uses the Philox PRNG, the state is controlled
- * by the counter, the salt and two keys.
- * If any of the keys and salt differ, the noise is
- * not correlated between two calls along the same counter
- * sequence.
+ * @tparam salt RNG salt
+ * @tparam N    Size of the noise vector
  *
+ * @return Vector of uniform random numbers.
  */
-template <RNGSalt salt>
-Utils::Vector3d v_noise(uint64_t counter, int key1, int key2 = 0) {
+template <RNGSalt salt, size_t N = 3,
+          typename std::enable_if<(N > 1) and (N <= 4), int>::type = 0>
+auto noise_uniform(uint64_t counter, int key1, int key2 = 0) {
+
+  auto const integers = philox_4_uint64s<salt>(counter, key1, key2);
+  Utils::VectorXd<N> noise{};
+  std::transform(integers.begin(), integers.begin() + N, noise.begin(),
+                 [](size_t value) { return Utils::uniform(value) - 0.5; });
+  return noise;
+}
 
-  auto const noise = philox_4_uint64s<salt>(counter, key1, key2);
+template <RNGSalt salt, size_t N,
+          typename std::enable_if<N == 1, int>::type = 0>
+auto noise_uniform(uint64_t counter, int key1, int key2 = 0) {
 
-  using Utils::uniform;
-  return Utils::Vector3d{uniform(noise[0]), uniform(noise[1]),
-                         uniform(noise[2])} -
-         Utils::Vector3d::broadcast(0.5);
+  auto const integers = philox_4_uint64s<salt>(counter, key1, key2);
```

Comment:
Wouldn't it be sufficient to use a `Philox2x64` here?

Reply:
`Philox2x64` can only take 1 particle id for the key:
```
src/core/random.hpp:95:18: error: too many initializers for ‘const key_type {aka const r123array1x64}’
   const key_type k{id1, id2};
```
Although the function that generates 1 random uniform value is only called in an NPT function that has no second particle (hence the second key is always 0), we could in the future have a need to generate 1 random value from two particle ids. We would also introduce ambiguity in the function call, since the function generating N=2-4 random values has an optional argument.

Reply:
It takes 64 bits as key, e.g. 2 * 32 bit or 2 int.